errs/error.go

```diff
@@ -299,7 +300,8 @@ func Errorf(code int, format string, args ...interface{}) error {
 // ApplyOptions applies the given options to the error if is the type *Error.
 // TODO(mariano): try to get rid of this.
 func ApplyOptions(err error, opts ...interface{}) error {
-	if e, ok := err.(*Error); ok {
+	var e *Error
+	if ok := errors.As(err, &e); ok {
```

Comment:
The ok here is not necessary.
```suggestion
	if errors.As(err, &e) {
```